File: frontend/src/app/core/models/courses-page.response.ts
export interface CourseOverview {
  id: string;
  code: string;
  title: string;
  prof: string;
  progress: number;
  nextDeadline: string;
  imageUrl: string;
}

export interface CompletedCourseSummary {
  title: string;
  date: string;
  grade: string;
}

export interface CoursesPageResponse {
  userName: string;
  userRole: string;
  userAvatarUrl: string;
  activeCourses: CourseOverview[];
  completedCourses: CompletedCourseSummary[];
}

File: backend/src/main/java/moodlev2/web/course/dto/CoursesPageResponse.java
package moodlev2.web.course.dto;

import java.util.List;

public record CoursesPageResponse(
        String userName,
        String userRole,
        String userAvatarUrl,
        List<CourseOverviewDto> activeCourses,
        List<CompletedCourseSummaryDto> completedCourses
) {}